app/script/auth/module/action/RuntimeAction.js

```diff
@@ -0,0 +1,124 @@
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see http://www.gnu.org/licenses/.
+ *
+ */
+
+import * as RuntimeActionCreator from './creator/RuntimeActionCreator';
+import RuntimeUtil from '../../util/RuntimeUtil';
+import Runtime from '../../Runtime';
+
+export function checkSupportedBrowser() {
+  const runtime = new Runtime();
+  return function(dispatch) {
+    if (runtime.isSupportedBrowser()) {
+      dispatch(RuntimeActionCreator.confirmSupportedBrowser());
+    }
+  };
+}
+
+export function checkIndexedDbSupport() {
+  return function(dispatch) {
+    dispatch(RuntimeActionCreator.startCheckIndexedDb());
+    hasIndexedDbSupport()
+      .then(() => {
+        dispatch(RuntimeActionCreator.finishCheckIndexedDb(true));
+      })
+      .catch(() => {
+        dispatch(RuntimeActionCreator.finishCheckIndexedDb(false));
+      });
+  };
+}
+
+export function checkCookieSupport() {
+  return function(dispatch) {
+    dispatch(RuntimeActionCreator.startCheckCookie());
+    hasCookieSupport()
+      .then(() => {
+        dispatch(RuntimeActionCreator.finishCheckCookie(true));
+      })
+      .catch(() => {
+        dispatch(RuntimeActionCreator.finishCheckCookie(false));
+      });
+  };
+}
+
+function hasCookieSupport() {
+  const cookieName = 'cookie_supported';
+
+  return new Promise((resolve, reject) => {
+    switch (navigator.cookieEnabled) {
+      case true:
+        return resolve();
+      case false:
+        return reject(new Error());
+      default:
+        Cookies.set(cookieName, 'yes');
+        if (Cookies.get(cookieName)) {
+          Cookies.remove(cookieName);
+          return resolve();
+        }
+        return reject(new Error());
+    }
+  });
+}
+
+function hasIndexedDbSupport() {
+  if (!window.indexedDB) {
+    return Promise.reject(new Error());
+  }
+
+  if (RuntimeUtil.isFirefox()) {
+    let dbOpenRequest;
+
+    try {
+      dbOpenRequest = window.indexedDB.open('test');
+      dbOpenRequest.onerror = event => {
+        if (dbOpenRequest.error) {
+          event.preventDefault();
+          Promise.reject(new Error());
+        }
+      };
+    } catch (error) {
+      return Promise.reject(new Error());
+    }
+
+    return new Promise((resolve, reject) => {
+      const interval = 10;
+      const maxRetry = 50;
+
+      function checkDbRequest(currentAttempt = 0) {
+        const tooManyAttempts = currentAttempt >= maxRetry;
+        const isRequestDone = dbOpenRequest.readyState === 'done';
+        const hasResult = !!dbOpenRequest.result;
+
+        if (isRequestDone && hasResult) {
+          return resolve();
+        }
+
+        if (tooManyAttempts || (isRequestDone && !hasResult)) {
+          return reject(new Error());
+        }
```

Comment:
I would move break this up differently for easier understanding of the logic:

````
if (isRequestDone) {
  return hasResult ? resolve() : reject(new Error('Failed to open IndexedDb'));
}

if (tooManyAttempts) {
  return reject(new Error('IndexedDb did not respond on open request within timeout'));
}
````